commitment/encoding.go

```diff
@@ -25,6 +25,12 @@ func AssetProofEncoder(w io.Writer, val any, buf *[8]byte) error {
 }
 
 func AssetProofDecoder(r io.Reader, val any, buf *[8]byte, l uint64) error {
+	// We currently only use this with tlv.DecodeP2P, but in case we ever
+	// don't, we still want to enforce a limit.
+	if l > tlv.MaxRecordSize {
+		return tlv.ErrRecordTooLarge
+	}
+
```

Comment:
Should we add a `if l == 0 {` check similar to what we have in `TapscriptPreimageDecoder`?

Reply:
No, since we don't do `l-1` which could then be negative. If `l` is `0` then `tlv.DVarBytes` will just read 0 bytes and the stream below will return `EOF`.